### PR TITLE
 对 ServiceDTO 重写equals、hashCode方法, ServiceController内魔数整理

### PR DESCRIPTION
**Pull request by hrsstd**
1. 对 ServiceDTO 重写equals、hashCode方法, 在ServiceController内 Set集合中使用保证数据唯一性。
 2. ServiceController 内部魔数定义

### CODE REVIEW

File: dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/common/AdminConstants.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    public final static String INTERROGATION_POINT = "?";
+
+    public static final String ANY_VALUE = "*";
+
+    public static final String APPLICATION = "application";
+
+    public static final String SERVICE = "service";
+
+    public static final String IP = "ip";
+}
```
Comment (nzomkxia) [inline dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/common/AdminConstants.java]: We already have a class: `org.apache.dubbo.admin.common.util.Constants`,  please add these variables there.

File: dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/config/ConfigCenter.java [hunk 2]
```diff
@@ -43,7 +43,7 @@
     @Value("${dubbo.configcenter:}")
     private String configCenter;
 
-    @Value("${dubbo.registry.address:}")
+    @Value("${dubbo.configcenter:}")
     private String registryAddress;
 
     private static String globalConfigPath = "config/dubbo/dubbo.properties";
```
Comment (nzomkxia) [inline dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/config/ConfigCenter.java]: both dubbo.configcenter and dubbo.registrey.address are necessary here, do not change

File: dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/controller/ServiceController.java [hunk 3]
```diff
@@ -52,33 +54,33 @@ public ServiceController(ProviderService providerService, ConsumerService consum
 
     @RequestMapping(method = RequestMethod.GET)
     public Set<ServiceDTO> searchService(@RequestParam String pattern,
-                                         @RequestParam String filter,@PathVariable String env) {
+        @RequestParam String filter,@PathVariable String env) {
```
Comment (nzomkxia) [inline dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/controller/ServiceController.java]: this format change is unnecessary

Comment (hrsstd) [reply dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/controller/ServiceController.java]: get

File: dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/config/ConfigCenter.java (hunk 2 above)
Comment (hrsstd) [reply dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/config/ConfigCenter.java]: if don't change this , can I change the  application.properties add a config ?
like this: dubbo.registry.address=zookeeper://127.0.0.1:2181

File: dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/common/AdminConstants.java (hunk 1 above)
Comment (hrsstd) [reply dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/common/AdminConstants.java]: ok , thank you. Later I will add these again.

**Comment by hrsstd** [comment]
Hi , nzomkxia ,I have to resubmit，please review again。

**Comment by hrsstd** [comment]
Choose English  as comment

**Pull request closed without merging**